File: lib/api/customer_api.dart
import 'dart:async';
import 'dart:io';
import 'dart:convert';
import 'package:dio/dio.dart';
import 'package:built_collection/built_collection.dart';
import 'package:built_value/serializer.dart';

import 'package:openapi/model/rest_result_of_customer_service_customer_address_response.dart';
import 'package:openapi/model/rest_result_of_customer_service_delete_address_response.dart';
import 'package:openapi/model/customer_service_customer_address_request.dart';
import 'package:openapi/model/customer_service_new_customer_default_address.dart';
import 'package:openapi/model/customer_service_new_customer_request.dart';
import 'package:openapi/model/rest_result_of_customer_service_customer_response.dart';
import 'package:openapi/model/customer_service_delete_address_request.dart';

class CustomerApi {
    final Dio _dio;
    Serializers _serializers;

    CustomerApi(this._dio, this._serializers);

        /// 
        ///
        /// 
        Future<Response<RestResultOfCustomerServiceDeleteAddressResponse>>apiCustomerDeleteAddressPost({ CustomerServiceDeleteAddressRequest customerServiceDeleteAddressRequest,CancelToken cancelToken, Map<String, String> headers,}) async {

        String _path = "/api/Customer/DeleteAddress";

        Map<String, dynamic> queryParams = {};
        Map<String, String> headerParams = Map.from(headers ?? {});
        dynamic bodyData;

        queryParams.removeWhere((key, value) => value == null);
        headerParams.removeWhere((key, value) => value == null);

        List<String> contentTypes = ["application/json","text/json","application/_*+json"];


            var serializedBody = _serializers.serialize(customerServiceDeleteAddressRequest);
            var jsoncustomerServiceDeleteAddressRequest = json.encode(serializedBody);
            bodyData = jsoncustomerServiceDeleteAddressRequest;

            return _dio.request(
            _path,
            queryParameters: queryParams,
            data: bodyData,
            options: Options(
            method: 'post'.toUpperCase(),
            headers: headerParams,
            extra: {
                'secure': [ {"type": "apiKey", "name": "Bearer", "keyName": "Authorization", "where": "header" }],
            },
            contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
            ),
            cancelToken: cancelToken,
            ).then((response) {

        var serializer = _serializers.serializerForType(RestResultOfCustomerServiceDeleteAddressResponse);
        var data = _serializers.deserializeWith<RestResultOfCustomerServiceDeleteAddressResponse>(serializer, response.data is String ? jsonDecode(response.data) : response.data);

            return Response<RestResultOfCustomerServiceDeleteAddressResponse>(
                data: data,
                headers: response.headers,
                request: response.request,
                redirects: response.redirects,
                statusCode: response.statusCode,
                statusMessage: response.statusMessage,
                extra: response.extra,
            );
            });
            }
        /// 
        ///
        /// 
        Future<Response<RestResultOfCustomerServiceCustomerAddressResponse>>apiCustomerGetCustomerAdressesPost({ CancelToken cancelToken, Map<String, String> headers,}) async {

        String _path = "/api/Customer/GetCustomerAdresses";

        Map<String, dynamic> queryParams = {};
        Map<String, String> headerParams = Map.from(headers ?? {});
        dynamic bodyData;

        queryParams.removeWhere((key, value) => value == null);
        headerParams.removeWhere((key, value) => value == null);

        List<String> contentTypes = [];



            return _dio.request(
            _path,
            queryParameters: queryParams,
            data: bodyData,
            options: Options(
            method: 'post'.toUpperCase(),
            headers: headerParams,
            extra: {
                'secure': [ {"type": "apiKey", "name": "Bearer", "keyName": "Authorization", "where": "header" }],
            },
            contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
            ),
            cancelToken: cancelToken,
            ).then((response) {

        var serializer = _serializers.serializerForType(RestResultOfCustomerServiceCustomerAddressResponse);
        var data = _serializers.deserializeWith<RestResultOfCustomerServiceCustomerAddressResponse>(serializer, response.data is String ? jsonDecode(response.data) : response.data);

            return Response<RestResultOfCustomerServiceCustomerAddressResponse>(
                data: data,
                headers: response.headers,
                request: response.request,
                redirects: response.redirects,
                statusCode: response.statusCode,
                statusMessage: response.statusMessage,
                extra: response.extra,
            );
            });
            }
        /// 
        ///
        /// 
        Future<Response<RestResultOfCustomerServiceCustomerResponse>>apiCustomerPost({ CancelToken cancelToken, Map<String, String> headers,}) async {

        String _path = "/api/Customer";

        Map<String, dynamic> queryParams = {};
        Map<String, String> headerParams = Map.from(headers ?? {});
        dynamic bodyData;

        queryParams.removeWhere((key, value) => value == null);
        headerParams.removeWhere((key, value) => value == null);

        List<String> contentTypes = [];



            return _dio.request(
            _path,
            queryParameters: queryParams,
            data: bodyData,
            options: Options(
            method: 'post'.toUpperCase(),
            headers: headerParams,
            extra: {
                'secure': [ {"type": "apiKey", "name": "Bearer", "keyName": "Authorization", "where": "header" }],
            },
            contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
            ),
            cancelToken: cancelToken,
            ).then((response) {

        var serializer = _serializers.serializerForType(RestResultOfCustomerServiceCustomerResponse);
        var data = _serializers.deserializeWith<RestResultOfCustomerServiceCustomerResponse>(serializer, response.data is String ? jsonDecode(response.data) : response.data);

            return Response<RestResultOfCustomerServiceCustomerResponse>(
                data: data,
                headers: response.headers,
                request: response.request,
                redirects: response.redirects,
                statusCode: response.statusCode,
                statusMessage: response.statusMessage,
                extra: response.extra,
            );
            });
            }
        /// 
        ///
        /// 
        Future<Response<RestResultOfCustomerServiceCustomerAddressResponse>>apiCustomerUpdateAddressPost({ CustomerServiceCustomerAddressRequest customerServiceCustomerAddressRequest,CancelToken cancelToken, Map<String, String> headers,}) async {

        String _path = "/api/Customer/UpdateAddress";

        Map<String, dynamic> queryParams = {};
        Map<String, String> headerParams = Map.from(headers ?? {});
        dynamic bodyData;

        queryParams.removeWhere((key, value) => value == null);
        headerParams.removeWhere((key, value) => value == null);

        List<String> contentTypes = ["application/json","text/json","application/_*+json"];


            var serializedBody = _serializers.serialize(customerServiceCustomerAddressRequest);
            var jsoncustomerServiceCustomerAddressRequest = json.encode(serializedBody);
            bodyData = jsoncustomerServiceCustomerAddressRequest;

            return _dio.request(
            _path,
            queryParameters: queryParams,
            data: bodyData,
            options: Options(
            method: 'post'.toUpperCase(),
            headers: headerParams,
            extra: {
                'secure': [ {"type": "apiKey", "name": "Bearer", "keyName": "Authorization", "where": "header" }],
            },
            contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
            ),
            cancelToken: cancelToken,
            ).then((response) {

        var serializer = _serializers.serializerForType(RestResultOfCustomerServiceCustomerAddressResponse);
        var data = _serializers.deserializeWith<RestResultOfCustomerServiceCustomerAddressResponse>(serializer, response.data is String ? jsonDecode(response.data) : response.data);

            return Response<RestResultOfCustomerServiceCustomerAddressResponse>(
                data: data,
                headers: response.headers,
                request: response.request,
                redirects: response.redirects,
                statusCode: response.statusCode,
                statusMessage: response.statusMessage,
                extra: response.extra,
            );
            });
            }
        /// 
        ///
        /// 
        Future<Response<RestResultOfCustomerServiceCustomerResponse>>apiCustomerUpdateCustomerDefultAddressPost({ CustomerServiceNewCustomerDefaultAddress customerServiceNewCustomerDefaultAddress,CancelToken cancelToken, Map<String, String> headers,}) async {

        String _path = "/api/Customer/UpdateCustomerDefultAddress";

        Map<String, dynamic> queryParams = {};
        Map<String, String> headerParams = Map.from(headers ?? {});
        dynamic bodyData;

        queryParams.removeWhere((key, value) => value == null);
        headerParams.removeWhere((key, value) => value == null);

        List<String> contentTypes = ["application/json","text/json","application/_*+json"];


            var serializedBody = _serializers.serialize(customerServiceNewCustomerDefaultAddress);
            var jsoncustomerServiceNewCustomerDefaultAddress = json.encode(serializedBody);
            bodyData = jsoncustomerServiceNewCustomerDefaultAddress;

            return _dio.request(
            _path,
            queryParameters: queryParams,
            data: bodyData,
            options: Options(
            method: 'post'.toUpperCase(),
            headers: headerParams,
            extra: {
                'secure': [ {"type": "apiKey", "name": "Bearer", "keyName": "Authorization", "where": "header" }],
            },
            contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
            ),
            cancelToken: cancelToken,
            ).then((response) {

        var serializer = _serializers.serializerForType(RestResultOfCustomerServiceCustomerResponse);
        var data = _serializers.deserializeWith<RestResultOfCustomerServiceCustomerResponse>(serializer, response.data is String ? jsonDecode(response.data) : response.data);

            return Response<RestResultOfCustomerServiceCustomerResponse>(
                data: data,
                headers: response.headers,
                request: response.request,
                redirects: response.redirects,
                statusCode: response.statusCode,
                statusMessage: response.statusMessage,
                extra: response.extra,
            );
            });
            }
        /// 
        ///
        /// 
        Future<Response<RestResultOfCustomerServiceCustomerResponse>>apiCustomerUpdateCustomerPost({ CustomerServiceNewCustomerRequest customerServiceNewCustomerRequest,CancelToken cancelToken, Map<String, String> headers,}) async {

        String _path = "/api/Customer/UpdateCustomer";

        Map<String, dynamic> queryParams = {};
        Map<String, String> headerParams = Map.from(headers ?? {});
        dynamic bodyData;

        queryParams.removeWhere((key, value) => value == null);
        headerParams.removeWhere((key, value) => value == null);

        List<String> contentTypes = ["application/json","text/json","application/_*+json"];


            var serializedBody = _serializers.serialize(customerServiceNewCustomerRequest);
            var jsoncustomerServiceNewCustomerRequest = json.encode(serializedBody);
            bodyData = jsoncustomerServiceNewCustomerRequest;

            return _dio.request(
            _path,
            queryParameters: queryParams,
            data: bodyData,
            options: Options(
            method: 'post'.toUpperCase(),
            headers: headerParams,
            extra: {
                'secure': [ {"type": "apiKey", "name": "Bearer", "keyName": "Authorization", "where": "header" }],
            },
            contentType: contentTypes.isNotEmpty ? contentTypes[0] : "application/json",
            ),
            cancelToken: cancelToken,
            ).then((response) {

        var serializer = _serializers.serializerForType(RestResultOfCustomerServiceCustomerResponse);
        var data = _serializers.deserializeWith<RestResultOfCustomerServiceCustomerResponse>(serializer, response.data is String ? jsonDecode(response.data) : response.data);

            return Response<RestResultOfCustomerServiceCustomerResponse>(
                data: data,
                headers: response.headers,
                request: response.request,
                redirects: response.redirects,
                statusCode: response.statusCode,
                statusMessage: response.statusMessage,
                extra: response.extra,
            );
            });
            }
        }

File: apis/CustomerApi.ts
// TODO: better import syntax?
import {BaseAPIRequestFactory, RequiredError, COLLECTION_FORMATS} from './baseapi';
import {Configuration} from '../configuration';
import {RequestContext, HttpMethod, ResponseContext, HttpFile} from '../http/http';
import {ObjectSerializer} from '../models/ObjectSerializer';
import {ApiException} from './exception';
import {canConsumeForm, isCodeInRange} from '../util';
import {SecurityAuthentication} from '../auth/auth';


import { CustomerServiceCustomerAddressRequest } from '../models/CustomerServiceCustomerAddressRequest';
import { CustomerServiceDeleteAddressRequest } from '../models/CustomerServiceDeleteAddressRequest';
import { CustomerServiceNewCustomerDefaultAddress } from '../models/CustomerServiceNewCustomerDefaultAddress';
import { CustomerServiceNewCustomerRequest } from '../models/CustomerServiceNewCustomerRequest';
import { RestResultOfCustomerServiceCustomerAddressResponse } from '../models/RestResultOfCustomerServiceCustomerAddressResponse';
import { RestResultOfCustomerServiceCustomerResponse } from '../models/RestResultOfCustomerServiceCustomerResponse';
import { RestResultOfCustomerServiceDeleteAddressResponse } from '../models/RestResultOfCustomerServiceDeleteAddressResponse';
import { RestResultOfListOfCustomerServiceCustomerMessageResponse } from '../models/RestResultOfListOfCustomerServiceCustomerMessageResponse';

/**
 * no description
 */
export class CustomerApiRequestFactory extends BaseAPIRequestFactory {

    /**
     * @param customerServiceDeleteAddressRequest 
     */
    public async apiCustomerDeleteAddressPost(customerServiceDeleteAddressRequest?: CustomerServiceDeleteAddressRequest, _options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;


        // Path Params
        const localVarPath = '/api/Customer/DeleteAddress';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")


        // Body Params
        const contentType = ObjectSerializer.getPreferredMediaType([
            "application/json",
        
            "text/json",
        
            "application/*+json"
        ]);
        requestContext.setHeaderParam("Content-Type", contentType);
        const serializedBody = ObjectSerializer.stringify(
            ObjectSerializer.serialize(customerServiceDeleteAddressRequest, "CustomerServiceDeleteAddressRequest", ""),
            contentType
        );
        requestContext.setBody(serializedBody);

        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

    /**
     */
    public async apiCustomerGetCustomerAdressesPost(_options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;

        // Path Params
        const localVarPath = '/api/Customer/GetCustomerAdresses';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")


        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

    /**
     * @param startSendDate 
     * @param endSendDate 
     */
    public async apiCustomerGetCustomerMessageGet(startSendDate?: Date, endSendDate?: Date, _options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;



        // Path Params
        const localVarPath = '/api/Customer/GetCustomerMessage';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.GET);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")

        // Query Params
        if (startSendDate !== undefined) {
            requestContext.setQueryParam("startSendDate", ObjectSerializer.serialize(startSendDate, "Date", "date-time"));
        }

        // Query Params
        if (endSendDate !== undefined) {
            requestContext.setQueryParam("endSendDate", ObjectSerializer.serialize(endSendDate, "Date", "date-time"));
        }


        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

    /**
     */
    public async apiCustomerPost(_options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;

        // Path Params
        const localVarPath = '/api/Customer';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")


        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

    /**
     * @param customerServiceCustomerAddressRequest 
     */
    public async apiCustomerUpdateAddressPost(customerServiceCustomerAddressRequest?: CustomerServiceCustomerAddressRequest, _options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;


        // Path Params
        const localVarPath = '/api/Customer/UpdateAddress';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")


        // Body Params
        const contentType = ObjectSerializer.getPreferredMediaType([
            "application/json",
        
            "text/json",
        
            "application/*+json"
        ]);
        requestContext.setHeaderParam("Content-Type", contentType);
        const serializedBody = ObjectSerializer.stringify(
            ObjectSerializer.serialize(customerServiceCustomerAddressRequest, "CustomerServiceCustomerAddressRequest", ""),
            contentType
        );
        requestContext.setBody(serializedBody);

        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

    /**
     * @param customerServiceNewCustomerDefaultAddress 
     */
    public async apiCustomerUpdateCustomerDefultAddressPost(customerServiceNewCustomerDefaultAddress?: CustomerServiceNewCustomerDefaultAddress, _options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;


        // Path Params
        const localVarPath = '/api/Customer/UpdateCustomerDefultAddress';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")


        // Body Params
        const contentType = ObjectSerializer.getPreferredMediaType([
            "application/json",
        
            "text/json",
        
            "application/*+json"
        ]);
        requestContext.setHeaderParam("Content-Type", contentType);
        const serializedBody = ObjectSerializer.stringify(
            ObjectSerializer.serialize(customerServiceNewCustomerDefaultAddress, "CustomerServiceNewCustomerDefaultAddress", ""),
            contentType
        );
        requestContext.setBody(serializedBody);

        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

    /**
     * @param customerServiceNewCustomerRequest 
     */
    public async apiCustomerUpdateCustomerPost(customerServiceNewCustomerRequest?: CustomerServiceNewCustomerRequest, _options?: Configuration): Promise<RequestContext> {
        let _config = _options || this.configuration;


        // Path Params
        const localVarPath = '/api/Customer/UpdateCustomer';

        // Make Request Context
        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.POST);
        requestContext.setHeaderParam("Accept", "application/json, */*;q=0.8")


        // Body Params
        const contentType = ObjectSerializer.getPreferredMediaType([
            "application/json",
        
            "text/json",
        
            "application/*+json"
        ]);
        requestContext.setHeaderParam("Content-Type", contentType);
        const serializedBody = ObjectSerializer.stringify(
            ObjectSerializer.serialize(customerServiceNewCustomerRequest, "CustomerServiceNewCustomerRequest", ""),
            contentType
        );
        requestContext.setBody(serializedBody);

        let authMethod: SecurityAuthentication | undefined;
        // Apply auth methods
        authMethod = _config.authMethods["Bearer"]
        if (authMethod?.applySecurityAuthentication) {
            await authMethod?.applySecurityAuthentication(requestContext);
        }
        
        const defaultAuth: SecurityAuthentication | undefined = _options?.authMethods?.default || this.configuration?.authMethods?.default
        if (defaultAuth?.applySecurityAuthentication) {
            await defaultAuth?.applySecurityAuthentication(requestContext);
        }

        return requestContext;
    }

}

export class CustomerApiResponseProcessor {

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerDeleteAddressPost
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerDeleteAddressPost(response: ResponseContext): Promise<RestResultOfCustomerServiceDeleteAddressResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfCustomerServiceDeleteAddressResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceDeleteAddressResponse", ""
            ) as RestResultOfCustomerServiceDeleteAddressResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfCustomerServiceDeleteAddressResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceDeleteAddressResponse", ""
            ) as RestResultOfCustomerServiceDeleteAddressResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerGetCustomerAdressesPost
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerGetCustomerAdressesPost(response: ResponseContext): Promise<RestResultOfCustomerServiceCustomerAddressResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfCustomerServiceCustomerAddressResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerAddressResponse", ""
            ) as RestResultOfCustomerServiceCustomerAddressResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfCustomerServiceCustomerAddressResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerAddressResponse", ""
            ) as RestResultOfCustomerServiceCustomerAddressResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerGetCustomerMessageGet
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerGetCustomerMessageGet(response: ResponseContext): Promise<RestResultOfListOfCustomerServiceCustomerMessageResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfListOfCustomerServiceCustomerMessageResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfListOfCustomerServiceCustomerMessageResponse", ""
            ) as RestResultOfListOfCustomerServiceCustomerMessageResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfListOfCustomerServiceCustomerMessageResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfListOfCustomerServiceCustomerMessageResponse", ""
            ) as RestResultOfListOfCustomerServiceCustomerMessageResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerPost
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerPost(response: ResponseContext): Promise<RestResultOfCustomerServiceCustomerResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfCustomerServiceCustomerResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerResponse", ""
            ) as RestResultOfCustomerServiceCustomerResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfCustomerServiceCustomerResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerResponse", ""
            ) as RestResultOfCustomerServiceCustomerResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerUpdateAddressPost
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerUpdateAddressPost(response: ResponseContext): Promise<RestResultOfCustomerServiceCustomerAddressResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfCustomerServiceCustomerAddressResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerAddressResponse", ""
            ) as RestResultOfCustomerServiceCustomerAddressResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfCustomerServiceCustomerAddressResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerAddressResponse", ""
            ) as RestResultOfCustomerServiceCustomerAddressResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerUpdateCustomerDefultAddressPost
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerUpdateCustomerDefultAddressPost(response: ResponseContext): Promise<RestResultOfCustomerServiceCustomerResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfCustomerServiceCustomerResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerResponse", ""
            ) as RestResultOfCustomerServiceCustomerResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfCustomerServiceCustomerResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerResponse", ""
            ) as RestResultOfCustomerServiceCustomerResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to apiCustomerUpdateCustomerPost
     * @throws ApiException if the response code was not in [200, 299]
     */
     public async apiCustomerUpdateCustomerPost(response: ResponseContext): Promise<RestResultOfCustomerServiceCustomerResponse > {
        const contentType = ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
        if (isCodeInRange("200", response.httpStatusCode)) {
            const body: RestResultOfCustomerServiceCustomerResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerResponse", ""
            ) as RestResultOfCustomerServiceCustomerResponse;
            return body;
        }

        // Work around for missing responses in specification, e.g. for petstore.yaml
        if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
            const body: RestResultOfCustomerServiceCustomerResponse = ObjectSerializer.deserialize(
                ObjectSerializer.parse(await response.body.text(), contentType),
                "RestResultOfCustomerServiceCustomerResponse", ""
            ) as RestResultOfCustomerServiceCustomerResponse;
            return body;
        }

        throw new ApiException<string | Blob | undefined>(response.httpStatusCode, "Unknown API Status Code!", await response.getBodyAsAny(), response.headers);
    }

}
